lib/delayed_job/delayed_job_callbacks.rb

```diff
@@ -8,21 +8,20 @@ class DelayedJobCallbacks < ::Delayed::Plugin
           block.call(job)
         ensure
           duration = Time.now - start
-          Metrician.gauge("queue.process", duration) if Metrician.configuration[:queue][:process][:enabled]
-          Metrician.gauge("#{job_metric_instrumentation_name(worker)}.process", duration) if Metrician.configuration[:queue][:job_specific][:enabled]
+          Metrician.gauge("jobs.run", duration) if Metrician.configuration[:jobs][:run][:enabled]
+          Metrician.gauge("jobs.run.#{job_metric_instrumentation_name(job)}", duration) if Metrician.configuration[:jobs][:job_specific][:enabled]
```

Comment:
I changed the order whilst making these changes because what you had proposed secretly was the opposite the advice we generally give people about how to structure a metric. i.e. least specific to most specific left-to-right. It also opposite of how I like to structure metrics, so… here we are.

I would make a small revision to the last list (task -> job):

```
jobs.run
jobs.error
jobs.run.queue.<queue_name>
jobs.error.queue.<queue_name>
jobs.run.job.<job_name>
jobs.error.job.<job_name>
jobs.run.queue.<queue_name>.job.<job_class>
jobs.error.queue<queue_name>.job.<job_class>
```

There is a small problem with the auto-complete, which *functionally* you can mostly get around by hitting enter on your keyboard. There is also a problem on the metrics explorer pages. The only way to get a graph of one of these metrics with sub metrics is to click on something and edit the URL, I think.

Maybe @tonydewan has some ideas about how to handle the autocomplete hinting, or is already working on something.